src/psyclone/parse/kernel.py

```diff
@@ -612,6 +612,81 @@ def get_kernel_metadata(name, ast):
     return ktype
 
 
+def getkerneldescriptors(name, ast, var_name='meta_args', var_type=None):
+    '''Get the required argument metadata information for a kernel.
+
+    :param str name: the name of the kernel (for error messages).
+    :param ast: metadata describing kernel arguments.
+    :type ast: :py:class:`fparser.one.block_statements.Type`
+    :param str var_name: the name of the variable storing the \
+           argument metadata. this argument is optional and defaults to \
```

Comment:
Our coding-style rules (https://psyclone-dev.readthedocs.io/en/latest/coding-style.html in case you're wondering) permit indenting to less than the start of the description in order to avoid having to introduce a lot of line breaks.